### PR TITLE
use jitdb@3 and where() operator

### PR DESCRIPTION
**Pull request by staltz**
Follow up to https://github.com/ssb-ngi-pointer/jitdb/pull/141. Also a **breaking change**.

### CODE REVIEW

**Comment by staltz** [comment]
No idea why tests keep failing, they are passing locally.

**Comment by arj03** [comment]
I also can't reproduce the error locally. Seems to be a sorting problem in jitdb. I'm tempted to rerun the tests here, but I'm not sure if that is going to change anything because both node 10 and 12 failed.

File: db.js [hunk 1]
```diff
@@ -392,18 +400,12 @@ exports.init = function (sbot, config) {
...
-    if (!first.meta) {
+    if (first.meta) {
+      return jitdbOperators.query(first, where(waitUntilReady), ...rest)
+    } else {
       const ops = fromDB(jitdb)
       ops.meta.db = this
-      return jitdbOperators.query(ops, and(waitUntilReady, first), ...rest)
-    } else {
-      return jitdbOperators.query(first, and(waitUntilReady), ...rest)
+      return jitdbOperators.query(ops, where(waitUntilReady), first, ...rest)
```
Comment (arj03) [inline db.js:408]: At first reading this, I thought this needed a `where(and(waitUntilReady, first)` but now I see that doesn't work.

File: indexes/full-mentions.js [hunk 2]
```diff
@@ -64,7 +64,7 @@ module.exports = class FullMentions extends Plugin {
             pl.read(this.level, Object.assign({}, opts, { live, old: false })),
             pull.map(parseInt10)
           )
-          cb(null, or(seqs(seqArr.map(parseInt10)), liveSeqs(ps))())
```
Comment (arj03) [inline indexes/full-mentions.js:67]: this is a nice improvement!

File: test/operators.js [hunk 3]
```diff
@@ -199,36 +199,7 @@ test('execute and(hasRoot(msgkey))', (t) => {
         t.error(err, 'no err')
 
         db.query(
```
Comment (arj03) [inline test/operators.js:201]: finally found something to nitpick, the test needs to be called `execute hasRoot(msgkey)` ;-)

Comment (arj03) [reply test/operators.js:201]: Pushed up a change for this to try and run the tests again :)

**Comment by arj03** [comment]
Well at least the failure is reproducible.

File: db.js (hunk 1 above)
Comment (staltz) [reply db.js:408]: Yeah, `first` could be `live` and `live` doesn't work inside `and` anymore (it used to).

**Comment by arj03** [comment]
This is really strange. It seems like there actually also was the same bug in the commit that bumped the version to 1.18.7. I just don't see why anything in bipf 1.5 should have triggered this problem. I'll try a debug branch.

**Review by arj03 — APPROVED** [review]

**Comment by arj03** [comment]
The error is unrelated to this PR. The error is randomly triggered because we basically add messages from the future and jitdb uses [min of arrival, declared](https://github.com/ssb-ngi-pointer/jitdb/blob/2ba4c90b558f5034a9e82a01e8f941a89d253bcc/index.js#L253). I'll fix the tests.

**Comment by staltz** [comment]
Great! Let's merge

**Pull request merged**